pkg/ccl/streamingccl/streamingest/stream_ingestion_dist_test.go

```diff
@@ -316,3 +316,139 @@ func TestSourceDestMatching(t *testing.T) {
+// constructSpanFrontierExecutionDetails. Refer to the method header for
+// details.
+func TestFrontierExecutionDetailFile(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	clearTimestamps := func(executionDetails []frontierExecutionDetails) []frontierExecutionDetails {
+		res := make([]frontierExecutionDetails, len(executionDetails))
+		for i, ed := range executionDetails {
+			ed.frontierTS = ""
+			ed.behindBy = ""
+			res[i] = ed
+		}
+		return res
+	}
+
+	for _, tc := range []struct {
+		name            string
+		partitionSpecs  execinfrapb.StreamIngestionPartitionSpecs
+		frontierEntries execinfrapb.FrontierEntries
+		expected        []frontierExecutionDetails
+	}{
+		{
+			name: "matching spans",
+			partitionSpecs: execinfrapb.StreamIngestionPartitionSpecs{
+				Specs: []*execinfrapb.StreamIngestionPartitionSpec{
+					{
+						SrcInstanceID:  1,
+						DestInstanceID: 2,
+						Spans: []roachpb.Span{
+							{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+						},
+					},
+				}},
+			frontierEntries: execinfrapb.FrontierEntries{ResolvedSpans: []jobspb.ResolvedSpan{
+				{
+					Span:      roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+					Timestamp: hlc.Timestamp{WallTime: 1},
+				},
+			}},
+			expected: []frontierExecutionDetails{
+				{
+					srcInstanceID:  1,
+					destInstanceID: 2,
+					span:           "{a-b}",
+				}},
+		},
+		{
+			name: "multi-partition",
+			partitionSpecs: execinfrapb.StreamIngestionPartitionSpecs{
+				Specs: []*execinfrapb.StreamIngestionPartitionSpec{
+					{
+						SrcInstanceID:  1,
+						DestInstanceID: 2,
+						Spans: []roachpb.Span{
+							{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+						},
+					},
+					{
+						SrcInstanceID:  1,
+						DestInstanceID: 3,
+						Spans: []roachpb.Span{
+							{Key: roachpb.Key("b"), EndKey: roachpb.Key("c")},
+						},
+					},
+				}},
+			frontierEntries: execinfrapb.FrontierEntries{ResolvedSpans: []jobspb.ResolvedSpan{
+				{
+					Span:      roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("a'")},
+					Timestamp: hlc.Timestamp{WallTime: 1},
+				},
+				{
+					Span:      roachpb.Span{Key: roachpb.Key("b"), EndKey: roachpb.Key("c")},
+					Timestamp: hlc.Timestamp{WallTime: 2},
+				},
+			}},
+			expected: []frontierExecutionDetails{
+				{
+					srcInstanceID:  1,
+					destInstanceID: 2,
+					span:           "a{-'}",
+				},
+				{
+					srcInstanceID:  1,
+					destInstanceID: 3,
+					span:           "{b-c}",
+				},
+			},
+		},
+		{
+			name: "merged frontier",
+			partitionSpecs: execinfrapb.StreamIngestionPartitionSpecs{
+				Specs: []*execinfrapb.StreamIngestionPartitionSpec{
+					{
+						SrcInstanceID:  1,
+						DestInstanceID: 2,
+						Spans: []roachpb.Span{
+							{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+						},
+					},
+					{
+						SrcInstanceID:  1,
+						DestInstanceID: 2,
+						Spans: []roachpb.Span{
+							{Key: roachpb.Key("b"), EndKey: roachpb.Key("d")},
```

Comment:
nit: i don't think this is a realistic test case: two different partition specs will never have the same src-dest pair. 

I could see two useful related test cases:
1.  Remove one of the partition specs, and add to the `Spans` list?
2. keep what you hav, and change one of the DestIDs to 3

Reply:
sounds good!